docs/worker-versioning.md

```diff
@@ -0,0 +1,229 @@
+```shell
+temporal task-queue versioning insert-assignment-rule --task-queue MY_TQ --build-id abc-123 --percentage 10
+```
+
+**Commit Build ID** to complete the rollout of "abc-123" and cleanup unnecessary rules.
+```shell
+temporal task-queue versioning commit-build-id --task-queue MY_TQ --build-id abc-123
+```
+
+**Add Redirect rule** from "abc-123" to "xyz-789":
+```shell
+temporal task-queue versioning add-redirect-rule --task-queue MY_TQ --source-build-id abc-123 --target-build-id xyz-789
+```
+
+**List Versioning rules**:
+```shell
+temporal task-queue versioning get-rules --task-queue MY_TQ
+```
+
+### Get Reachability Info via CLI
+You can use `temporal task-queue describe` command to get reachability status of a Build ID.
+Example: report reachability of Build ID "abc-123" on Task Queue MY_TQ:
+```shell
+temporal task-queue describe --task-queue MY_TQ --select-build-id abc-123 --report-reachability
+```
+
+The reachability status can be one of the following:
+- **REACHABLE:** Build ID may be used by new workflows or activities (base on versioning 
+rules), or there MAY be open workflows or backlogged activities assigned to it.
+- **CLOSED_WORKFLOWS_ONLY:** Build ID does not have open workflows and is not reachable
+by new workflows, but MAY have closed workflows within the namespace retention period. 
+Not applicable to activity-only task queues.
+- **UNREACHABLE:** Build ID is not used for new executions, nor it has been used by 
+any existing execution within the retention period.
+- **UNSPECIFIED:** Task reachability is not reported
+
+#### Caveats:
+- Task Reachability is eventually consistent; there may be a delay until it converges to the most
+accurate value but it is designed in a way to take the more conservative side until it converges.
+For example REACHABLE is more conservative than CLOSED_WORKFLOWS_ONLY.
+
+- Future activities who inherit their workflow's Build ID but not its Task Queue will not be
+accounted for reachability as server cannot know if they'll happen as they do not use
+assignment rules of their Task Queue. Same goes for Child Workflows or Continue-As-New Workflows
+who inherit the parent/previous workflow's Build ID but not its Task Queue. In those cases, make
+sure to query reachability for the parent/previous workflow's Task Queue as well.
+
+## Best Practices
+[WIP]
+
+## Planned Improvements
+Worker Versioning is currently in Pre-Release (alpha) stage. We continue to improve this
+feature before the public release. Here are some improvements planned for the coming months:
+
+- [TBD]
+
+## Migration Notes
+### Migrating from Unversioned Task Queue
+To migrate your existing Task Queue, follow the normal procedure as if you upgrade from
+one Build ID to another.
+
+The only limitation is that, as of now, redirect rules with unversioned source is not 
+supported. Hence, if you want to redirect you long-running unversioned Workflows to a 
+Build ID that is not possible. (This may change in the future.)
```

Comment:
Hmm, doesn't this mean people with long-running workflows can't even get started?  If so, we should headline that toward the top that workers with long-running workflows are currently unsupported.
As a reader, I want to know upfront, "should I be using this?"  As the docs exist today, I have to read the whole doc and make some subtle inferences to realize this.

Reply:
They can leave the unversioned ones on unversioned workers and start new ones on versioned workers. Essentially treating "unversioned" as "version 0".

They can't redirect but they can upgrade their unversioned workers with new (compatible) code, just as they could do without versioning.

Reply:
But that means they'd need to have two deployments indefinitely.